spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableScanRDD.scala

```diff
@@ -221,10 +238,36 @@ class CassandraTableScanRDD[R] private[connector](
     if (containsPartitionKey(where)) {
       CassandraPartitionGenerator(connector, tableDef, Some(1), splitSize)
     } else {
-      CassandraPartitionGenerator(connector, tableDef, splitCount, splitSize)
+      CassandraPartitionGenerator(connector, tableDef, numPartitions.orElse(splitCount), splitSize)
     }
   }
 
+  /**
+    * The method does not create CoalesceRDD but reduce number of parittions to read from Cassandra.
```

Comment:
Maybe reword

"This method overrides the default spark behavior and will not create a CoalesceRDD. Instead it will reduce the number of partitions by adjusting the partitioning of C* data on read. Using this method will override spark.cassandra.input.split.size."